### PR TITLE
NGMX-1536 refactor for Bob's API change

### PR DESCRIPTION
**Pull request by mbrookhart**
### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/ngraph/ngraph_nnvm_utils.h [hunk 1]
```diff
@@ -49,7 +49,9 @@ inline std::shared_ptr<TensorView> NDArray_to_TensorView(
   auto shape = TShape_to_NShape(input.shape());
   const auto& element_type = getType(input.dtype());
 
-  auto TV = backend->make_primary_tensor_view(element_type, shape);
+  // TODO(mbrookhart): I don't think Ashok's memory sharing PR has a
+  // create_tensor implementation? this will probably conflict
```
Comment (ashokei) [inline src/ngraph/ngraph_nnvm_utils.h]: currently , both API are supported, so there should be no conflict. I have a sprint task to update ngraph tensorview api for this change, and also look into layout issue.

Comment (mbrookhart) [reply src/ngraph/ngraph_nnvm_utils.h]: Thanks!

**Comment by mbrookhart** [comment]
TESTNOW

**Comment by lnguyen-nvn** [comment]
@mbrookhart : Ready to merge ?
